### ISSUE TITLE
`bsearch` and `bs` disappear after pressing space or select

### ISSUE THREAD

**Issue by chenspc** [drgrib/alfred-bear#56]
This might be a basic question but I haven't been able to find a solution anywhere online. I wondered if anyone knows what was going on here. Thanks. 

I can use `bn` (+ argument) and `bnew` (+ argument) keywords without a problem. If I type `bcs`, Alfred would initially give the correct suggestion. 
<img width="764" alt="bcs_before_space" src="https://github.com/drgrib/alfred-bear/assets/10161227/a1390ea5-d02d-4c8a-a5e4-c7d2ed3d8607">

However, as soon as I press `SPACE` or `TAB` before getting a chance to type in the search argument, the Bear related workflow options would disappear. 
<img width="764" alt="bcs_after_space" src="https://github.com/drgrib/alfred-bear/assets/10161227/e9b020af-1dc8-4161-af7f-b25458b7ca30">

The same happens to `bcsearch`, `bs`, `bsearch` too. For `bsearch`, it happens when I type the letter `h`.
<img width="764" alt="bsearch_before_h" src="https://github.com/drgrib/alfred-bear/assets/10161227/5024ade5-9ea1-485c-9b7f-3982787c039d">

<img width="764" alt="bsearch_after_h" src="https://github.com/drgrib/alfred-bear/assets/10161227/9d748022-69d1-41b6-a616-51cb5329ddf5">

**Comment by drgrib** [comment]
You are probably getting an error for the workflow. Try opening the debug for it under Workflows > Bear and see what message you get when you type `bs` or `bcs`.

![debug](https://github.com/drgrib/alfred-bear/assets/3879367/ad82b6c9-c97e-4935-96fd-568e16c5f498)

**Comment by chenspc** [comment]
Thanks for your response. I've tried what you suggested by typing `bs abc` and `bcs abc`, and got the following log

```
[02:04:36.670] Bear[Script Filter] Queuing argument ''
[02:04:36.906] Bear[Script Filter] Script with argv '' finished
[02:04:36.912] ERROR: Bear[Script Filter] Code 2: panic: no such table: Z_7TAGS

goroutine 1 [running]:
main.main()
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/search/main.go:29 +0x5e9
[02:04:43.243] Bear[Script Filter] Queuing argument 'a'
[02:04:43.331] Bear[Script Filter] Script with argv 'a' finished
[02:04:43.335] ERROR: Bear[Script Filter] Code 2: panic: no such table: Z_7TAGS

goroutine 1 [running]:
main.main()
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/search/main.go:29 +0x5e9
[02:04:43.363] Bear[Script Filter] Queuing argument 'ab'
[02:04:43.487] Bear[Script Filter] Script with argv 'ab' finished
[02:04:43.493] ERROR: Bear[Script Filter] Code 2: panic: no such table: Z_7TAGS

goroutine 1 [running]:
main.main()
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/search/main.go:29 +0x5e9
[02:04:43.656] Bear[Script Filter] Queuing argument 'abc'
[02:04:43.742] Bear[Script Filter] Script with argv 'abc' finished
[02:04:43.745] ERROR: Bear[Script Filter] Code 2: panic: no such table: Z_7TAGS

goroutine 1 [running]:
main.main()
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/search/main.go:29 +0x5e9
[02:07:31.670] Logging Started...
[02:07:42.107] Bear[Script Filter] Queuing argument ''
[02:07:42.350] Bear[Script Filter] Script with argv '' finished
[02:07:42.362] ERROR: Bear[Script Filter] Code 1: no such table: Z_7TAGS
github.com/drgrib/alfred-bear/db.LiteDB.Query
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:176
github.com/drgrib/alfred-bear/core.GetSearchRows
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/core/core.go:193
main.main
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/csearch/main.go:31
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
github.com/drgrib/alfred-bear/core.GetSearchRows
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/core/core.go:195
main.main
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/csearch/main.go:31
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
[02:07:44.625] Bear[Script Filter] Queuing argument 'a'
[02:07:44.740] Bear[Script Filter] Queuing argument 'ab'
[02:07:44.775] Bear[Script Filter] Script with argv 'a' finished
[02:07:44.785] ERROR: Bear[Script Filter] Code 1: no such table: Z_7TAGS
github.com/drgrib/alfred-bear/db.LiteDB.Query
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:176
github.com/drgrib/alfred-bear/db.LiteDB.QueryNotesByText.func1
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:264
github.com/drgrib/alfred-bear/db.multiWordQuery
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:310
github.com/drgrib/alfred-bear/db.LiteDB.QueryNotesByText
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:266
github.com/drgrib/alfred-bear/core.GetSearchRows
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/core/core.go:207
main.main
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/csearch/main.go:31
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
github.com/drgrib/alfred-bear/core.GetSearchRows
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/core/core.go:209
main.main
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/csearch/main.go:31
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
[02:07:44.891] Bear[Script Filter] Script with argv 'ab' finished
[02:07:44.898] ERROR: Bear[Script Filter] Code 1: no such table: Z_7TAGS
github.com/drgrib/alfred-bear/db.LiteDB.Query
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:176
github.com/drgrib/alfred-bear/db.LiteDB.QueryNotesByText.func1
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:264
github.com/drgrib/alfred-bear/db.multiWordQuery
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:310
github.com/drgrib/alfred-bear/db.LiteDB.QueryNotesByText
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:266
github.com/drgrib/alfred-bear/core.GetSearchRows
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/core/core.go:207
main.main
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/csearch/main.go:31
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
github.com/drgrib/alfred-bear/core.GetSearchRows
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/core/core.go:209
main.main
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/csearch/main.go:31
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
[02:07:45.049] Bear[Script Filter] Queuing argument 'abc'
[02:07:45.201] Bear[Script Filter] Script with argv 'abc' finished
[02:07:45.211] ERROR: Bear[Script Filter] Code 1: no such table: Z_7TAGS
github.com/drgrib/alfred-bear/db.LiteDB.Query
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:176
github.com/drgrib/alfred-bear/db.LiteDB.QueryNotesByText.func1
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:264
github.com/drgrib/alfred-bear/db.multiWordQuery
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:310
github.com/drgrib/alfred-bear/db.LiteDB.QueryNotesByText
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/db/db.go:266
github.com/drgrib/alfred-bear/core.GetSearchRows
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/core/core.go:207
main.main
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/csearch/main.go:31
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
github.com/drgrib/alfred-bear/core.GetSearchRows
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/core/core.go:209
main.main
	/Users/credford/Dropbox/Alfred/Alfred.alfredpreferences/workflows/user.workflow.48B0F8E6-2BC8-4EB9-B35E-1DBB820AA8E7/cmd/csearch/main.go:31
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
```

**Comment by drgrib** [comment]
I see. It sounds like you are using Bear v2. Did you get the beta version? I won't be releasing support for it until it is public and finalized and I can test it with #54.

**Comment by chenspc** [comment]
Ah yes, I indeed switched to Bear 2 beta recently. I think I'll just be patient and wait for the full release. 

Thanks again for making the workflow. It looks awesome! Looking forward to v2.

**Comment by drgrib** [comment]
Glad it looks useful to you! I'm looking forward to v2 as well.

**Comment by JasperRobinson** [comment]
tl;dr—👍 Bear2 workflow fully functional. 

[Follow up from email, last week] 

I had installed the Bear2 workflow, last week, but 'bs' would not work, even with the quarantine authorisation. De-bugging while running yielded:

```
[10:55:18.055] Logging Started...
[10:55:23.639] Bear[[Script Filter](alfredpreferences:workflows%3Eworkflow%3Euser.workflow.6FC9E059-46E3-4200-90F2-745362247D62%3E11295A3D-C960-41A3-9670-DFCBDF58C9D9)] Queuing argument ''
[10:55:23.659] Bear[[Script Filter](alfredpreferences:workflows%3Eworkflow%3Euser.workflow.6FC9E059-46E3-4200-90F2-745362247D62%3E11295A3D-C960-41A3-9670-DFCBDF58C9D9)] Script with argv '' finished
[10:55:23.663] ERROR: Bear[[Script Filter](alfredpreferences:workflows%3Eworkflow%3Euser.workflow.6FC9E059-46E3-4200-90F2-745362247D62%3E11295A3D-C960-41A3-9670-DFCBDF58C9D9)] Code 126: /Users/lpzjw/Library/Caches/com.runningwithcrayons.Alfred/Workflow Scripts/1006B4B6-134A-4E50-9B80-72A0347C89FB: line 1: cmd/search/search: Operation not permitted
```

I deleted the Bear2 workflow from Alfred and re-downloaded it. Now it's fine. Last week I did not first delete the original Bear workflow. Maybe something from the original Bear workflow remained..?
